### PR TITLE
Add Diffblue Deeptest tests

### PR DESCRIPTION
**Pull request by petersear**
Pre-PR test review request.
These tests are also available to view at:
http://cx.dashboard.diffblue.com/Mybatis-PageHelper/1/tests

**PR text:**

The test cases contained in this PR were automatically generated by [Diffblue](http://www.diffblue.com/)'s [Deeptest](http://www.diffblue.com/papers/overview.pdf) software. 

Diffblue is an Oxford-based start-up that aims to automate traditional coding tasks to increase developer productivity. One of our missions is to provide useful tests to open source projects. In return we would appreciate your feedback about our tests to help us further improve our tool. 
    
These tests are part of a batch of 102 in total, providing an increase of 9% in line coverage.
We would be happy to provide the remaining tests via a subsequent PR.

Results for various open source projects can be viewed at:
    [http://diff.blue/demo](http://diff.blue/demo)

We welcome any feedback you have on these tests.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -103,6 +103,33 @@
...
+        <dependency>
+            <groupId>org.powermock</groupId>
+            <artifactId>powermock-api-mockito</artifactId>
+            <version>1.6.5</version>
+        </dependency>
+        <dependency>
+            <groupId>org.powermock</groupId>
+            <artifactId>powermock-api-mockito-common</artifactId>
+            <version>1.6.5</version>
+        </dependency>
```
Comment (paulkennethkent) [inline pom.xml]: powermock-api-mockito-common is not required, there is a PR to change this.

File: src/test/java/com/github/pagehelper/dialect/AbstractRowBoundsDialectTest.java [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+  /*
+   * Test generated by Diffblue Deeptest.
+   * This test case covers:
+   *  - conditional line 50 branch to line 51
+   *  - conditional line 51 branch to line 51
+   *  - conditional line 52 branch to line 52
+   */
+
+  @Test
+  public void beforeCountInputNotNullZeroNotNullOutputFalse() throws InvocationTargetException {
```
Comment (paulkennethkent) [inline src/test/java/com/github/pagehelper/dialect/AbstractRowBoundsDialectTest.java]: The already have a test that covers this line according to the online demo. Might be worth leaving out.

File: src/test/java/com/github/pagehelper/dialect/AbstractRowBoundsDialectTest.java [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+  /*
+   * Test generated by Diffblue Deeptest.
+   * This test case covers:
+   *  - conditional line 50 branch to line 51
+   *  - conditional line 51 branch to line 51
+   *  - conditional line 52 branch to line 52
+   */
+
+  @Test
+  public void beforeCountInputNotNullZeroNotNullOutputTrue() throws InvocationTargetException {
```
Comment (paulkennethkent) [inline src/test/java/com/github/pagehelper/dialect/AbstractRowBoundsDialectTest.java]: The already have a test that covers this line according to the online demo. Might be worth leaving out.

File: src/test/java/com/github/pagehelper/util/ExecutorUtilTest.java [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+
+  /* testedClasses: ExecutorUtil */
+
+  /*
+   * Test generated by Diffblue Deeptest.
+   * This test case covers the entire method.
+   */
+  @PrepareForTest({Configuration.class})
+  @Test
+  public void getExistedMappedStatementInputNotNullNotNullOutputNotNull()
```
Comment (paulkennethkent) [inline src/test/java/com/github/pagehelper/util/ExecutorUtilTest.java]: The already have a test that covers this line according to the online demo. Even though there is a single line missing, I think this is down to bluecov.

**Review by paulkennethkent — CHANGES_REQUESTED** [review]
Remove tests that are more than likely covered by existing tests and change pom.xml

**Review by peterschrammel — COMMENTED** [review]
None of these tests significantly above borderline, unfortunately.

**Comment by peterschrammel** [comment]
What's the existing coverage?

**Comment by petersear** [comment]
These tests mostly come from the invalid bucket from test-validator which identified 9 "valid" tests out of 102.
One of those failed to compile and 3 others appear to cover lines which are already covered (as pointed out by @paulkennethkent). That was the reason I rejected some of the other "valid" tests.
Existing coverage was calculated by platform to be around 65%.

File: pom.xml (hunk 1 above)
Comment (petersear) [reply pom.xml]: Done, thanks.

File: src/test/java/com/github/pagehelper/dialect/AbstractRowBoundsDialectTest.java (hunk 2 above)
Comment (petersear) [reply src/test/java/com/github/pagehelper/dialect/AbstractRowBoundsDialectTest.java]: Removed.

File: src/test/java/com/github/pagehelper/dialect/AbstractRowBoundsDialectTest.java (hunk 3 above)
Comment (petersear) [reply src/test/java/com/github/pagehelper/dialect/AbstractRowBoundsDialectTest.java]: Removed.

File: src/test/java/com/github/pagehelper/util/ExecutorUtilTest.java (hunk 4 above)
Comment (petersear) [reply src/test/java/com/github/pagehelper/util/ExecutorUtilTest.java]: Agreed. Removed.

**Comment by paulkennethkent** [comment]
Close PR, no good tests this time around 👎

**Review by paulkennethkent — COMMENTED** [review]
👎

**Comment by petersear** [comment]
No suitable tests found. Closing this PR.

**Pull request closed without merging**